dmscripts/notify_suppliers_with_incomplete_applications.py

```diff
@@ -0,0 +1,56 @@
+from dmapiclient import DataAPIClient
+from dmscripts.helpers import logging_helpers
+from dmscripts.helpers.auth_helpers import get_auth_token
+from dmscripts.helpers.email_helpers import scripts_notify_client
+from dmscripts.helpers.logging_helpers import logging
+from dmutils.email.exceptions import EmailError
+from dmutils.email.helpers import hash_string
+from dmutils.env_helpers import get_api_endpoint_from_stage
+
+
+NOTIFY_TEMPLATE_ID = '25c7c763-fe51-418f-8e51-130c391edc35'
+
+
+def notify_suppliers_with_incomplete_applications(doc_opt_arguments):
+    dry_run = doc_opt_arguments['--dry-run']
+    framework = doc_opt_arguments['<framework>']
+    stage = doc_opt_arguments['<stage>']
+
+    logger = logging_helpers.configure_logger({"dmapiclient": logging.INFO})
+    mail_client = scripts_notify_client(doc_opt_arguments['<notify_api_key>'], logger=logger)
+    data_api_client = DataAPIClient(
+        base_url=get_api_endpoint_from_stage(stage), auth_token=get_auth_token('api', stage)
+    )
+
+    prefix = "[Dry Run] " if dry_run else ""
+    error_count = 0
+
+    for sf in data_api_client.find_framework_suppliers_iter(framework):
+        message = ''
+        if not sf.get('applicationCompanyDetailsConfirmed', None):
+            message += 'confirm your company details\n'
+        if not (sf.get('declaration', {})):
+            message += 'finish your supplier declaration\n'
+        if not (data_api_client.find_draft_services(sf['supplierId'], framework=framework).get('services', [])):
+            message += 'mark at least one of your services as complete\n'
```

Comment:
We don't get to put markdown in Notify personalizations do we? If so, we could make these bullet points, otherwise it'll have to do.